src/corehost/cli/comhost/exports.cpp

```diff
@@ -0,0 +1,250 @@
+// Copyright (c) .NET Foundation and contributors. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+#include "comhost.h"
+#include <corehost.h>
+#include <error_codes.h>
+#include <trace.h>
+#include <type_traits>
+
+using comhost::clsid_map_entry;
+using comhost::clsid_map;
+
+#if defined(_WIN32)
+
+// COM entry points are defined without the __declspec(dllexport) attribute.
+// The issue here is that the compiler will throw an error regarding linkage
+// redefinion. The solution here is to the use a .def file on Windows.
+#define COM_API extern "C"
+
+#else
+
+#define COM_API SHARED_API
+
+#endif // _WIN32
+
+COM_API HRESULT STDMETHODCALLTYPE DllGetClassObject(
+    _In_ REFCLSID rclsid,
+    _In_ REFIID riid,
+    _Outptr_ LPVOID FAR* ppv)
+{
+    // Check if the CLSID map contains a mapping
+    clsid_map map;
+    RETURN_HRESULT_IF_EXCEPT(map = comhost::get_clsid_map());
```

Comment:
Do we foresee a world where reading the json every time could either 1) produce inconsistent results or 2) be a performance issue?

Reply:
1) No.
2) Quite possibly. I am more than willing to cache that but until we can start measuring perf in this scenario I am apprehensive to optimize some of these paths beyond obvious paths that are idempotent.